src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/OracleEngine.java

```diff
@@ -162,17 +226,40 @@ protected int entityToPreparedStatement(final DbEntity entity, final PreparedSta
                 }
                 switch (column.getDbColumnType()) {
                     case BLOB:
-                        ps.setBytesForBlob(i, objectToArray(val));
+                        final byte[] valArray = objectToArray(val);
+                        if (fromBatch && valArray.length > MIN_SIZE_FOR_BLOB) {
+                            // Use a blob for columns greater than 4K when inside a batch
+                            // update because the Oracle driver creates one and only releases
+                            // it when the PreparedStatement is closed. This will be closed
+                            // explicitly when the batch is flushed.
+                            final Blob blob = ps.getConnection().createBlob();
+                            blob.setBytes(1, objectToArray(val));
+                            ps.setBlob(i, blob);
+                            postFlushActions.add(blob::free);
+                        } else {
+                            ps.setBytesForBlob(i, valArray);
+                        }
                         break;
                     case JSON:
                     case CLOB:
                         if (val == null) {
                             ps.setNull(i, Types.CLOB);
                             break;
                         }
-
                         if (val instanceof String) {
-                            ps.setStringForClob(i, (String) val);
+                            final String valString = (String) val;
+                            if (fromBatch && valString.length() > MIN_SIZE_FOR_CLOB) {
+                                // Use a clob for columns greater than 35K when inside a batch
+                                // update because the Oracle driver creates one and only releases
+                                // it when the PreparedStatement is closed. This will be closed
+                                // explicitly when the batch is flushed.
+                                final Clob clob = ps.getConnection().createClob();
+                                clob.setString(1, (String) val);
```

Comment:
```suggestion
                                clob.setString(1, valString);
```
same as above (even though in this case there is no performance difference)